### PR TITLE
Add a DB option to require (and leverage) explicit transaction boundaries

### PR DESCRIPTION
**Pull request by jimfulton**
See https://github.com/zodb/relstorage/issues/147

### CODE REVIEW

File: src/ZODB/DB.py [hunk 1]
```diff
@@ -363,6 +363,7 @@ def __init__(self,
                  databases=None,
                  xrefs=True,
                  large_record_size=1<<24,
+                 explicit_transactions = False,
```
Comment (freddrake) [inline src/ZODB/DB.py]: What's with the inconsistent spacing?

Comment (jimfulton) [inline src/ZODB/DB.py]: Sigh, I missed that. This is one of the places where PEP 8 (and Guildo) are wrong.

**Comment by jimfulton** [comment]
This might be simplified if I implemented the proposal I made on the transaction list yesterday.

Rather than configuring this via the DB. Connection objects could get their explicitness via the transaction managers passed to them.

**Comment by jimfulton** [comment]
Yeah, I think it would be better to control this via the TM.

**Pull request closed without merging**